x-pack/legacy/plugins/snapshot_restore/public/app/lib/authorization/components/auth_privileges.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useContext } from 'react';
+
+import { AuthorizationContext, MissingPrivileges } from './authorization_provider';
+
+interface Props {
+  /**
+   * Each required privilege must have the format "section.privilege".
+   * To indicate that *all* privileges from a section are required, we can use the asterix
+   * e.g. "index.*"
+   */
+  requiredPrivileges: string | string[];
+  children: (childrenProps: {
+    isLoading: boolean;
+    hasPrivileges: boolean;
+    requiredPrivilegesMissing: MissingPrivileges;
+  }) => JSX.Element;
+}
+
+type Privilege = [string, string];
+
+const toArray = (value: string | string[]): string[] =>
+  Array.isArray(value) ? (value as string[]) : ([value] as string[]);
+
+export const AuthPrivileges = ({ requiredPrivileges, children }: Props) => {
```

Comment:
Nit: `AuthorizationPrivileges` or simply "Authorization" would be more consistent with our other names, and removes any ambiguity around what auth stands for (authorization or authentication).

Reply:
Ok, I will use `AuthorizationPrivileges`.

"Authorization" and "Privileges" are 2 different things 😊 The "Authorization" is the _mechanism_ in place that will handle a user "privileges". It could, for example, also manage "roles" and would still be called "Authorization". 
Here, the component is handling "privileges" so I'll change it to `AuthorizationPrivileges`.

Reply:
Edit: I went back to my original `<WithPrivileges>` as it is what best describes this component. I just saw this convention used by the app search team.

```js
<WithPrivileges privileges="index.*">
  {/* anything here requires the above privileges */}
</WithPrivileges>
```

cc @jen-huang